fem/pfespace.cpp

```diff
@@ -147,20 +149,37 @@ void ParFiniteElementSpace::Construct()
       // cut space.
       ConstructTrueDofs();
 
+      ngedofs = ngfdofs = 0;
+      gfdofs = NULL;
+
       // calculate number of ghost DOFs
       ngvdofs = pncmesh->GetNGhostVertices()
                 * fec->DofForGeometry(Geometry::POINT);
 
-      ngedofs = ngfdofs = 0;
       if (pmesh->Dimension() > 1)
       {
          ngedofs = pncmesh->GetNGhostEdges()
                    * fec->DofForGeometry(Geometry::SEGMENT);
       }
+
       if (pmesh->Dimension() > 2)
       {
-         ngfdofs = pncmesh->GetNGhostFaces()
-                   * fec->DofForGeometry(pncmesh->GetGhostFaceGeometry(0));
+         if (fdofs != NULL) // have mixed faces
```

Comment:
I believe `fdofs` is allocated whenever face dofs exist in this space not only when the mesh has a mixture of face types.  You may need to use `Mesh::HasGeometry` to explicitly check for the presence of triangle and quadrilateral faces.

Reply:
The comment may not be 100% precise, I can remove it. The code however should be OK, the reasoning is that if `fdofs != NULL`, it has to be used, no matter what the reason is behind it being non-NULL. That said, if all faces have the same geometry, it would be strange to have `fdofs` allocated, so maybe it's a bug elsewhere?

Reply:
You make a good point.  Perhaps we need to take a closer look at `fdofs`.  Not necessarily as part of this PR but at some point.